src/renderer/actions/install.js

```diff
@@ -129,59 +147,141 @@ export default async function(config) {
+            // installation's main xdg-config directory, and install BetterDiscord there.
+            // Before: "/home/foo/.var/app/com.discordapp.Discord/config/discord/0.0.19/modules/discord_desktop_core".
+            // After: "/home/foo/.var/app/com.discordapp.Discord/config/BetterDiscord".
+            // NOTE: Installation paths on Linux always contain those 4 suffix parts.
+            // NOTE: The requirement that BetterDiscord is installed in xdg-config
+            // is hardcoded into betterdiscord.asar, so we can't change these paths.
+            bdFolders.push(path.join(discordPath, "../../../..", "BetterDiscord"));
+        }
+    }
 
+    if (bdFolders.length < 1) {
+        lognewline("No Discord paths provided.");
+        return fail();
+    }
+
+
+    // Create BetterDiscord's directory hierarchies.
     lognewline("Creating required directories...");
-    const makeDirErr = await makeDirectories(bdFolder, bdDataFolder, bdThemesFolder, bdPluginsFolder);
-    if (makeDirErr) return fail();
-    log("✅ Directories created");
+    for (const bdBaseFolder of bdFolders) {
+        const makeDirErr = await makeDirectories(
+            bdBaseFolder,
+            path.join(bdBaseFolder, "data"),
+            path.join(bdBaseFolder, "plugins"),
+            path.join(bdBaseFolder, "themes")
+        );
+        if (makeDirErr) return fail();
+    }
+    log("✅ Directories created.");
     progress.set(MAKE_DIR_PROGRESS);
 
 
-    if (process.platform === "darwin") {
-        lognewline("Checking for old MacOS installation...");
-        const found = await checkOldMacOS(oldBDFolder);
+    // Migrate old BetterDiscord data to the new locations.
+    if (legacyBDFolders.length >= 1) {
+        const platformName = (process.platform === "win32" ? "Windows" : (process.platform === "darwin" ? "macOS" : "Linux"));
+        lognewline(`Checking for old ${platformName} installations...`);
         progress.set(CHECK_OLD_INSTALL);
-        if (found) {
-            const confirmation = await remote.dialog.showMessageBox(remote.BrowserWindow.getFocusedWindow(), {
-                type: "question",
-                title: "Old Install Found",
-                message: "Found an old BD installation, do you want to transfer your plugins and themes?",
-                noLink: true,
-                cancelId: 1,
-                buttons: ["Yes", "No"]
-            });
-        
-            if (confirmation.response === 0) {
-                await transferOldAddons(path.join(oldBDFolder, "plugins"), path.join(bdFolder, "plugins"));
-                await transferOldAddons(path.join(oldBDFolder, "themes"), path.join(bdFolder, "themes"));
-                progress.set(TRANSFER_OLD_ADDONS);
+        for (const oldBDFolder of legacyBDFolders) {
+            const found = await checkOldBDFolder(oldBDFolder);
+            if (found) {
+                // NOTE: We migrate the data to the first new folder, which due
```

Comment:
Please convert long comments to multiline format

Reply:
Alright, will do! :)